lib/fencing/st_client.c

```diff
@@ -1427,7 +1437,7 @@ stonith_send_notification(gpointer data, gpointer user_data)
         return;
     }
 
-    st_event = xml_to_event(blob->xml);
+    st_event = xml_to_event(blob->xml, &result);
```

Comment:
why don't we  create the result inside xml_to_event?
maybe I see the reason with later changes ...

Reply:
xml_to_event() sets event->opaque = result, so the memory has to still be valid just after this when the callback is invoked. stonith_send_notification() frees the event immediately after that, so that function scope is sufficient. It might be less brittle if we dynamically allocated result in xml_to_event() and freed it in event_free() ...

Reply:
or allocate it in one chunk together with the event
or have it static